### PR TITLE
Gb9 stories

### PR DESCRIPTION
**Pull request by kris-kolosek**

### CODE REVIEW

File: spec/features/user_spec.rb [hunk 1]
```diff
@@ -43,4 +43,15 @@
       expect(current_path).to eql(user_session_path)
     end
   end
+
+  describe "#admin?" do
+    let! :each do
+      @user = create(:admin_role)
+      login_as(@user)
+    end
+
```
Comment (sniffler) [inline spec/features/user_spec.rb]: Trailing whitespace detected.

File: db/migrate/20180119145644_devise_invitable_add_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      t.index      :invitations_count
+      t.index      :invitation_token, unique: true # for invitable
+      t.index      :invited_by_id
+    end
+  end
+
+  def down
+    change_table :users do |t|
+      t.remove_references :invited_by, polymorphic: true
+      t.remove :invitations_count, :invitation_limit, :invitation_sent_at, :invitation_accepted_at, :invitation_token, :invitation_created_at
```
Comment (sniffler) [inline db/migrate/20180119145644_devise_invitable_add_to_users.rb]: Line is too long. [141/80]

File: app/services/text_formatter.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+require "redcarpet"
+
+module TextFormatter
+
```
Comment (sniffler) [inline app/services/text_formatter.rb]: Trailing whitespace detected.

File: app/models/user.rb [hunk 4]
```diff
@@ -17,6 +18,10 @@ class User < ApplicationRecord
   private
 
   def set_role
-    self.role = Role.new(user: self, role: Role::ADMIN)
+    if User.count.zero?
```
Comment (sniffler) [inline app/models/user.rb]: Use the return of the conditional for variable assignment and comparison.

File: app/controllers/stories_controller.rb [hunk 5]
```diff
@@ -0,0 +1,61 @@
+class StoriesController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_story, only: %i(show edit update destroy)
+
+  def index
+    @stories = Story.view_premissions(current_user).order(updated_at: :desc).filter(params[:filter], current_user)
```
Comment (sniffler) [inline app/controllers/stories_controller.rb]: Line is too long. [114/80]

File: app/controllers/invitations_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      format.html do
+        super
+      end
+    end
+  end
+
+  protected
+
+  def update_sanitized_params
+    devise_parameter_sanitizer.permit(:accept_invitation, keys: [:password, :invitation_token])
```
Comment (sniffler) [inline app/controllers/invitations_controller.rb]: Use %i or %I for an array of symbols.<br>Line is too long. [95/80]

File: app/controllers/invitations_controller.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  before_action :update_sanitized_params, only: :update
+
+  # PUT /resource/invitation
+  def update
+    respond_to do |format|
+      format.js do
+        invitation_token = Devise.token_generator.digest(resource_class, :invitation_token, update_resource_params[:invitation_token])
+        self.resource = resource_class.where(invitation_token: invitation_token).first
+        resource.skip_password = true
+        resource.update_attributes update_resource_params.except(:invitation_token)
```
Comment (sniffler) [inline app/controllers/invitations_controller.rb]: Line is too long. [83/80]

File: app/controllers/invitations_controller.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
+class InvitationsController < Devise::InvitationsController
+  before_action :update_sanitized_params, only: :update
+
+  # PUT /resource/invitation
+  def update
+    respond_to do |format|
+      format.js do
+        invitation_token = Devise.token_generator.digest(resource_class, :invitation_token, update_resource_params[:invitation_token])
+        self.resource = resource_class.where(invitation_token: invitation_token).first
```
Comment (sniffler) [inline app/controllers/invitations_controller.rb]: Line is too long. [86/80]

File: app/controllers/invitations_controller.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+class InvitationsController < Devise::InvitationsController
+  before_action :update_sanitized_params, only: :update
+
+  # PUT /resource/invitation
+  def update
+    respond_to do |format|
+      format.js do
+        invitation_token = Devise.token_generator.digest(resource_class, :invitation_token, update_resource_params[:invitation_token])
```
Comment (sniffler) [inline app/controllers/invitations_controller.rb]: Line is too long. [134/80]

File: app/assets/stylesheets/pages/_story.scss [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+  padding: 24px 4vw 3vw;
+  position: absolute;
+  width: 100%;
+
+  .form-title {
+    color: $black;
+    display: block;
+    font-size: 35px;
+    font-weight: 500;
+    letter-spacing: .8px;
```
Comment (sniffler) [inline app/assets/stylesheets/pages/_story.scss]: `.8` should be written with a leading zero as `0.8`

File: app/services/text_formatter.rb [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
```
Comment (sniffler) [inline app/services/text_formatter.rb]: Extra empty line detected at module body beginning.

File: app/assets/stylesheets/pages/_story.scss [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+.edit-story-content {
+  background-color: $white;
+  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Oxygen,Ubuntu,Cantarell,Droid Sans,Helvetica Neue,sans-serif;
+  left: 300px;
+  max-width: 1200px;
+  padding: 24px 4vw 3vw;
+  position: absolute;
+  width: 100%;
+
+  .form-title {
```
Comment (sniffler) [inline app/assets/stylesheets/pages/_story.scss]: Rule set contains (11/10) properties

File: app/assets/stylesheets/pages/_story.scss [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+      font-size: 20px;
+      font-weight: 700;
+      letter-spacing: .8px;
+      line-height: 1.15em;
+      min-height: auto;
+      overflow: hidden;
+      width: 100%;
+    }
+
+    .form-text {
```
Comment (sniffler) [inline app/assets/stylesheets/pages/_story.scss]: Selector should have depth of applicability no greater than 2, but was 3

File: app/controllers/stories_controller.rb [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+    else
+      flash[:error] = "Error while creating new story"
+      render action: :new
+    end
+  end
+
+  def update
+    if @story.update(story_params)
+      flash[:success] = "Story #{@story.title} is successfully updated."
```
Comment (salac-kolosek) [inline app/controllers/stories_controller.rb]: You can define add_flash_types :success, :error in application controller and then use it like redirect_to team_members_path, success: "Profile is removed successfully" for example.

File: app/models/story.rb [hunk 15]
```diff
@@ -0,0 +1,20 @@
+class Story < ApplicationRecord
+  belongs_to :user
+  before_save :markdown
+
+  def markdown
```
Comment (salac-kolosek) [inline app/models/story.rb]: put markdown in private methods.

File: config/initializers/devise.rb [hunk 16]
```diff
@@ -14,7 +14,7 @@
   # Configure the e-mail address which will be shown in Devise::Mailer,
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+  config.mailer_sender = 'kris-ghost@kolosek.com'
```
Comment (salac-kolosek) [inline config/initializers/devise.rb]: Put this in env variable?

File: app/models/story.rb (hunk 15 above)
Comment (kris-kolosek) [reply app/models/story.rb]: If i put markdown as private method, I will not be able to call it in view.

File: config/initializers/devise.rb (hunk 16 above)
Comment (kris-kolosek) [reply config/initializers/devise.rb]: Yes, I will. Thanks for pointing it out :)

File: app/models/story.rb (hunk 15 above)
Comment (kolosek) [reply app/models/story.rb]: far reply.

**Pull request closed without merging**